controllers/user.controller.js

```diff
@@ -13,3 +22,172 @@ exports.getAllUsers = async (req, res, next) => {
     });
   }
 };
+
+// @desc    Get User by query params
+// @route   GET /api/v1/users/get?field=value&....
+exports.getUsersByQuery = async (req, res, next) => {
+  var users = null;
```

Comment:
why do we need to define var users as null, we can directly use const user in try block

Reply:
i guess both will work same

Reply:
Yes they will.